src/metabase/search/filter.clj

```diff
@@ -113,6 +114,41 @@
   [_filter model query creator-id]
   (sql.helpers/where query (default-created-by-fitler-clause model creator-id)))
 
+;; Verified filters
+
+(defmethod build-optional-filter-query [:verified "card"]
+  [_filter model query verified]
+  ;; TODO: should be :content-verification once the feature flag project is merged
+  (premium-features/assert-has-feature :content-verification (deferred-tru "Content Verification"))
+  (if verified
+    (-> query
+        (sql.helpers/join :moderation_review
+                          [:= :moderation_review.moderated_item_id
+                           (search.config/column-with-model-alias model :id)])
+        (sql.helpers/where [:= :moderation_review.status "verified"]
+                           [:= :moderation_review.moderated_item_type "card"]
+                           [:= :moderation_review.most_recent true]))
+    (-> query
```

Comment:
The falsey branch feels incorrect. The original product doc says users should have the ability to search for "Only verified items" or "All items". It shouldn't be possible to search for "Only non-verified items". I think a suitable implementation would be to make `verified=false` mean that no filter is applied.

Reply:
yea I thought about that but we discussed and agreed on this approach.

https://metaboat.slack.com/archives/C057T1QTB3L/p1690344682110089

if you don't want the filter to be applied, then `verified` should not be present or have value = nil

Reply:
Where in that discussion did others agree having `verified=false` should filter out verified items?

From Conor:
>  let’s go with the implied behavior in the product doc e.g. you can’t search for only “non-verified” items.

The conclusion is there's no significant use-case where someone wants “please show me all non-verified items".

Reply:
I mean we'll still offer that option, it's just how we implement it and @oisincoveney is okay with having `nil` means: return everything.

Reply:
Yes. Then since Oisin doesn't mind, _I_ disagree with your proposal to make `verified=false` mean "return all non-verified items". It's just extra code to maintain and test, which we don't need

Reply:
You can even tighten up the schema and make `verified :- [:maybe true?]`

Reply:
Yep, from the discussion on Slack I had the impression that we wouldn't do the "get non-verified items" piece. Usually I would say "add it for flexibility later" but if it's going to be an issue in terms of maintainability then we should stick with the original proposal for now

Reply:
okay I'm onboard. Updated so that it could be either `true` or `nil` https://github.com/metabase/metabase/pull/32662/commits/bde1b754eb6488e7491b996bd946ecb2cdd098d2